src/test/java/com/coveo/spillway/functional/SpillwayFunctionalTests.java

```diff
@@ -101,14 +101,14 @@ public void oneMillionConcurrentRequestsWith100Threads() throws InterruptedExcep
     threadPool.shutdown();
     threadPool.awaitTermination(1, TimeUnit.MINUTES);
 
-    assertThat(counter.get()).isEqualTo(ONE_MILLION);
+    assertThat(counter.get()).isAtMost(ONE_MILLION + 10);
```

Comment:
Non-deterministic tests are the worst ;P

I say either ensure that we can have a precise count or have a lower and upper bound that is based on maths.